### PR TITLE
fix #2818: throw an error when a function assignment has duplicate parameter names

### LINKED ISSUE #2818: `math.evaluate("f(x,x)=x")` should throw an error
### Discussed in https://github.com/josdejong/mathjs/discussions/2815

<div type='discussions-op-text'>

<sup>Originally posted by **ChristopherChudzicki** October 16, 2022</sup>
**Describe the bug**
`f(x,x) = x` parses and evaluates to a function with somewhat unexpected behavior.

**To Reproduce**
Run:
```js
const math = require('mathjs')
const f = math.evaluate("f(x, x) = x")
f(1, 2)
```

I came across this while working on (the rewrite of) my app, math3d.org. See https://www.math3d.org/S9Kp3MDh5 for how this came up. I want user input of `f(x, x)` to be an error. It seems like something unintended/confusing by users. Certainly I can handle this as a consumer of the mathjs library.

But... how should MathJs handle this? Options:

1. The current behavior is correct: allow parsing and evluating `f(x, x) = ...`, with later parameter values taking precedence.
2. Throw an error during parsing
3. Throw an error during evaluation: can't determine which value of x is correct. 

(2) Seems too restrictive. Parsing the expression can be useful for getting structural information.  (1) and (3) both seem somewhat reasonable... Curious if this has come up before / is intended behavior. </div>

### PR DESCRIPTION
**Pull request by josdejong**
Fixes #2818

@ChristopherChudzicki can you review this PR?

### CODE REVIEW

File: test/unit-tests/expression/node/FunctionAssignmentNode.test.js [hunk 1]
```diff
@@ -302,6 +302,20 @@ describe('FunctionAssignmentNode', function () {
     }, /Callback function must return a Node/)
   })
 
+  it('should throw an error when having duplicate variables', function () {
```
Comment (ChristopherChudzicki) [inline test/unit-tests/expression/node/FunctionAssignmentNode.test.js:305]: I noticed
```ts
  const name = typeof param === 'string' ? param : param.name
```
up above.

Do you think it's worth adding a test case like
```ts
assert.throws(function () {
      console.log(new FunctionAssignmentNode('f', ['x', { name: 'x' }], new ConstantNode(2)))
    }, new Error('Duplicate parameter name "x"'))
```
that uses the `{ name: ... }` version?

**Review by ChristopherChudzicki — APPROVED** [review]
Looks good to me. Left one suggestion 👍 

Thanks for the quick fix!

File: test/unit-tests/expression/node/FunctionAssignmentNode.test.js (hunk 1 above)
Comment (josdejong) [reply test/unit-tests/expression/node/FunctionAssignmentNode.test.js:305]: That's a good idea, thanks!

Done

**Comment by josdejong** [comment]
Fix published now in `v11.3.1`

**Pull request merged**